lib/bitpagos/client.rb

```diff
@@ -22,42 +22,67 @@ def get_transaction(transaction_id)
       retrieve_transactions(nil, transaction_id)
     end
 
-    def all_transactions
-      retrieve_transactions
+    # Retrieve all transactions, or specify them with parameters, to retrieve
+    # allow retrieving paginated transactions.
+    #
+    # @param [Hash] Allows to specify transaction type, limit or offset
+    # @option [Symbol] status (:pending, :waiting, :completed, :partially_paid)
+    # @option [Integer] offset
+    # @option [Integer] limit
+    # @return [Hash]
+    def transactions(params = {})
+      if status = params[:status]
+        params[:status] = get_transaction_type_from_symbol(status)
+      end
+      retrieve_transactions(params)
     end
 
     def completed_transactions
-      retrieve_transactions(COMPLETED)
+      retrieve_transactions({status: COMPLETED})
     end
 
     def waiting_transactions
-      retrieve_transactions(WAITING)
+      retrieve_transactions({status: WAITING})
     end
 
     def pending_transactions
-      retrieve_transactions(PENDING)
+      retrieve_transactions({status: PENDING})
```

Comment:
@mauro-oto The curly braces are not necessary when you write lines like this one. I'd recommend you get rid of them. ;)